app2/src/lib/transfer/normal/steps/ApprovalStep.svelte

```diff
@@ -8,22 +8,13 @@ import { getCosmWasmClient } from "$lib/services/cosmos/clients.ts"
 import { getWalletClient } from "$lib/services/evm/clients.ts"
 import { wallets } from "$lib/stores/wallets.svelte.ts"
 import type { Steps } from "$lib/transfer/normal/steps"
-import {
-  hasFailedExit as cosmosHasFailedExit,
-  isComplete as cosmosIsComplete,
-  nextStateCosmos,
-  TransactionSubmissionCosmos,
-} from "$lib/transfer/shared/services/write-cosmos.ts"
-import {
-  hasFailedExit as evmHasFailedExit,
-  isComplete as evmIsComplete,
-  nextStateEvm,
-  TransactionSubmissionEvm,
-} from "$lib/transfer/shared/services/write-evm.ts"
+import * as WriteCosmos from "$lib/transfer/shared/services/write-cosmos.ts"
+import * as WriteEvm from "$lib/transfer/shared/services/write-evm.ts"
```

Comment:
Had to rewrite to ensure type safety such that we can make strong guarantees about error handling.